etl/assets.py

```diff
@@ -1,71 +1,96 @@
 import json
-import os
-from pathlib import Path
 
-from dagster import asset, define_asset_job
-
-from etl.resources import (
-    OpenAiEmbeddingModelResource,
-    OpenAiGenerativeModelResource,
-    WikipediaReaderResource,
+from dagster import asset
+from langchain_core.vectorstores import VectorStore
+
+from etl.embedding_model_pipelines import OpenAiEmbeddingModelPipeline
+from etl.generative_model_pipelines import OpenAiGenerativeModelPipeline
+from etl.models import (
+    DocumentTuple,
+    OpenAiPipelineConfig,
+    OpenAiSettings,
+    RecordTuple,
+    data_files_config_from_env_vars,
+    output_config_from_env_vars,
 )
+from etl.readers import WikipediaReader
+from etl.utils import create_documents
 
 
 @asset
-def wikipedia_articles_from_storage(
-    wikipedia_reader_resource: WikipediaReaderResource,
-):
+def wikipedia_articles_from_storage() -> RecordTuple:
     """Materialize an asset of Wikipedia articles."""
 
-    return tuple(wikipedia_reader_resource.read())
+    return RecordTuple(
+        records=tuple(
+            WikipediaReader(data_files_config=data_files_config_from_env_vars).read()
+        )
+    )
 
 
 @asset
 def wikipedia_articles_with_summaries(
-    wikipedia_articles_from_storage,
-    openai_generative_model_resource: OpenAiGenerativeModelResource,
-):
+    wikipedia_articles_from_storage: RecordTuple, config: OpenAiPipelineConfig
+) -> RecordTuple:
     """Materialize an asset of Wikipedia articles with summaries."""
 
-    enriched_wikipedia_articles = [
-        openai_generative_model_resource.enrich_record(wikipedia_article)
-        for wikipedia_article in wikipedia_articles_from_storage
-    ]
+    return RecordTuple(
+        records=tuple(
+            OpenAiGenerativeModelPipeline(config).enrich_record(wikipedia_article)
+            for wikipedia_article in wikipedia_articles_from_storage.records
+        )
+    )
+
+
+@asset
+def wikipedia_articles_with_summaries_json_file(
+    wikipedia_articles_with_summaries: RecordTuple,
+) -> None:
+    """Store the asset of Wikipedia articles with summaries as JSON."""
+
+    output_directory_path = output_config_from_env_vars.parse().directory_path
 
-    output_directory = Path(__file__).parent.absolute() / "data" / "output"
-    if not output_directory.exists():
-        os.mkdir(output_directory)
+    output_directory_path.mkdir(exist_ok=True)
 
-    enriched_wikipedia_output_directory = (
-        output_directory / "wikipedia_articles_with_summaries.txt"
+    enriched_wikipedia_output_file_path = (
```

Comment:
I'd put these as derived properties on output config rather than hard-coding any file names at all.

In other words, the resource can have a single directory, and the Parsed object can just expose a  computed `@property` for each file name.

Reply:
This should have a `.jsonl` extension if it's JSON lines.

Reply:
The properties do look much better than hard-coding. I am using them for directories as well